ledger/db/src/test_utils/mock_ledger.rs

```diff
@@ -265,11 +266,127 @@ pub fn get_test_ledger_blocks(n_blocks: usize) -> Vec<(Block, BlockContents)> {
     blocks_and_contents
 }
 
+/// Creates test blocks with customized parameters
+pub fn get_custom_test_ledger_blocks(
+    outputs_per_recipient_per_block: usize,
+    num_accounts: usize,
+    num_blocks: usize,
+    key_images_per_block: usize,
+    max_token_id: u32,
+) -> Vec<(Block, BlockContents)> {
+    let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+
+    let num_outputs: u64 =
+        (num_accounts * outputs_per_recipient_per_block * num_blocks * (max_token_id as usize + 1))
+            as u64;
+    assert!(num_outputs >= 16);
+
+    let picomob_per_output: u64 = (TOTAL_MOB / num_outputs) * 1_000_000_000_000;
+    let recipients = vec![AccountKey::random(&mut rng).default_subaddress(); num_accounts];
```

Comment:
This repeats the same address; it seems better to generate a bunch of addresses, e.g. 
```suggestion
    let recipients = (0..num_accounts).map(|_| AccountKey::random(&mut rng).default_subaddress()).collect::<Vec<_>>();
```